OpenDreamShared/Compiler/DM/DMParser.cs

```diff
@@ -1016,6 +1017,47 @@ public partial class DMParser : Parser<Token> {
             return null;
         }
 
+        public DMASTProcStatementTryCatch TryCatch() {
+            if (Check(TokenType.DM_Try)) {
+                DMASTProcBlockInner tryBody = ProcBlock();
+                if (tryBody == null) {
+                    DMASTProcStatement statement = ProcStatement();
+
+                    if (statement == null) Error("Expected body or statement");
+                    tryBody = new DMASTProcBlockInner(new DMASTProcStatement[] { statement });
+                }
+
+                Warning("Exceptions in 'try/catch' blocks are currently not caught");
+
+                Newline();
+                Consume(TokenType.DM_Catch, "Expected catch");
+
+                // catch(var/exception/E)
+                // TODO: handle correctly
+                DMASTDefinitionParameter[] parameters = null;
+                if (Check(TokenType.DM_LeftParenthesis))
+                {
+                    BracketWhitespace();
+                    parameters = DefinitionParameters();
```

Comment:
You cannot have more than one var definition in a catch. It may be better to instead parse this with `ProcVarDeclaration(allowMultiple: false)`